core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/udf/pythonV2/PythonUDFOpDescV2.scala

```diff
@@ -65,36 +68,68 @@ class PythonUDFOpDescV2 extends OperatorDescriptor {
 
   override def operatorExecutor(operatorSchemaInfo: OperatorSchemaInfo) = {
     Preconditions.checkArgument(workers >= 1, "Need at least 1 worker.", Array())
+    val opInfo = this.operatorInfo
+    val partitionRequirement: List[Option[PartitionInfo]] = if (inputPorts != null) {
+      inputPorts.map(p => Option(p.partitionRequirement))
```

Comment:
There is one thing not handled that prevents this from executing with 0 input ports. With 0 input ports, the operator should be treated as a source operator. Amber handles source operators differently from regular operators that the controller sends a "Start" message to source operators only.

Reply:
After a deeper thought, let's make this a TODO in a future PR. This may require significant changes.